webapp/src/immersivereader.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from "react";
+import * as data from "./data";
+import * as auth from "./auth";
+import * as core from "./core";
+import * as sui from "./sui";
+import * as ImmersiveReader from '@microsoft/immersive-reader-sdk';
+
+import Cloud = pxt.Cloud;
+
+export type ImmersiveReaderToken = {
+    token: string;
+    subdomain: string;
+    expiration: number;
+    expiresIn: number;
+}
+
+interface ImmersiveReaderProps {
+    content: string;
+}
+
+
+function beautifyText(content: string): string {
+    const cleaningFuncs = [cleanImages, cleanBlockAnnotation, cleanBold];
+    let contentWIP = content;
+
+    cleaningFuncs.forEach(clean => {
+        contentWIP = clean(contentWIP);
+    })
+
+    return contentWIP;
+
+    // Change ``|| around blocks to {}
+    function cleanBlockAnnotation(content: string): string {
+        const blockLiteral = /``\|\|([\w|\s]+:[\w|\s]+)\|\|``/
+        const blockRegex = new RegExp(blockLiteral, "g");
+        const blocksRemoved = content.replace(blockRegex, (blockDef) => {
+            const match = blockDef.match(blockLiteral);
```

Comment:
Shouldn't need to rematch in here -- you should get the matched groups as parameters to this fn https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_function_as_a_parameter

Reply:
:O